### ISSUE TITLE
Config::Vars more specific value False gets overridden by less specific non-False

### ISSUE THREAD

**Issue by raunz** [ytti/oxidized#1893]
There's an issue with lib/oxidized/config/vars.rb, when trying to set specific value for Node intentionally to False and there's a less specific variable set to True. Result is True, not False as expected. 
Found this when trying to set remove_secret to False for some specific scenarios, while the model default is true.

```
/etc/oxidized/config
models:
  fortios:
    vars:
      remove_secret: true

node = Oxidized::Node.new(name: "samplenode", ip: "127.0.0.1", group: nil, model: "fortios", output: 'git', vars: { remove_secret: false })

node.vars :remove_secret  returns true.
```
Also if a group variable is intentionally False, it will be overridden by Model variable True value.

**Comment by ytti** [comment]
You are right.
https://github.com/ytti/oxidized/blob/master/lib/oxidized/config/vars.rb#L5

This line would set it to false. Then we continue to set it if it's false or nil.

Probably refactored one would use returns. Something like

```ruby
def vars(name)
   return @node.vars[name] if @node.vars && @node.vars.has_key?(name)
   return Oxidized.config.groups[@node.group].vars[name] if Oxidized.config.groups.has_key?(@node.group) && Oxidized.config.groups[@node.group].vars.has_key?(name)
    return Oxidized.config.models[@node.model.class.name.to_s.downcase].vars[name] if Oxidized.config.models.has_key(@node.model.class.name.to_s.downcase) && Oxidized.config.models[@node.model.class.name.to_s.downcase].vars.has_key?(name)
    return Oxidized.config.vars[name] if Oxidized.config.vars.has_key?(name)
  end
```

Didn't check, even for typos.

**Comment by raunz** [comment]
Resolved with commit 42aa7d17b755e536cc4d948b7a525acf020fe5b5
